### PR TITLE
Add notes about running Beats on Openshift

### PR DESCRIPTION
**Pull request by barkbay**
This PR adds some notes and an example to explain how to run Beats on Openshift.

### CODE REVIEW

File: docs/advanced-topics/openshift.asciidoc [hunk 1]
```diff
@@ -226,3 +227,70 @@ elasticsearch-sample-es-6qk52mz5jk scc:restricted
 elasticsearch-sample-es-dg4vvpm2mr scc:restricted
 kibana-sample-kb-97c6b6b8d-lqfd2 scc:restricted
 ----
+
+[id="{p}-openshift-beats"]
+== Grant privileged permissions to Beats
+
+Deploying Beats on Openshift may require some privileged permissions. This section describes how to create a ServiceAccount, add the ServiceAccount to the `privileged` SCC and then use it to run Beats.
```
Comment (alaudazzi) [inline docs/advanced-topics/openshift.asciidoc:234]:
```suggestion
Deploying Beats on Openshift may require some privileged permissions. This section describes how to create a ServiceAccount, add the ServiceAccount to the `privileged` SCC, and use it to run Beats.
```

File: docs/advanced-topics/openshift.asciidoc [hunk 2]
```diff
@@ -226,3 +227,70 @@ elasticsearch-sample-es-6qk52mz5jk scc:restricted
...
+
+The following example assumes that Beats is deployed in a Namespace named `elastic` and we want to run Beats with a ServiceAccount named `filebeat`, replace these values according to your environment.
```
Comment (alaudazzi) [inline docs/advanced-topics/openshift.asciidoc:236]:
```suggestion
The following example assumes that Beats is deployed in the Namespace `elastic` with the ServiceAccount `filebeat`. You can replace these values according to your environment.
```

File: docs/advanced-topics/openshift.asciidoc [hunk 3]
```diff
@@ -226,3 +227,70 @@ elasticsearch-sample-es-6qk52mz5jk scc:restricted
...
+
+NOTE: If you have used the examples from the link:{eck_github}/tree/{eck_release_branch}/config/recipes/beats[recipes directory] the ServiceAccount may already exist.
```
Comment (alaudazzi) [inline docs/advanced-topics/openshift.asciidoc:238]:
```suggestion
NOTE: If you used the examples from the link:{eck_github}/tree/{eck_release_branch}/config/recipes/beats[recipes directory], the ServiceAccount may already exist.
```

File: docs/advanced-topics/openshift.asciidoc [hunk 4]
```diff
@@ -226,3 +227,70 @@ elasticsearch-sample-es-6qk52mz5jk scc:restricted
...
+----
+
+If SELinux is enabled the Beat Pod may fail with the following message:
+
+[source,shell]
+----
+Exiting: Failed to create Beat meta file: open /usr/share/heartbeat/data/meta.json.new: permission denied
+----
+
+In order to fix this error the label `svirt_sandbox_file_t` must be applied to the directory `/var/lib/elastic/heartbeat/heartbeat-data/` on the Kubernetes node:
```
Comment (alaudazzi) [inline docs/advanced-topics/openshift.asciidoc:289]:
```suggestion
To fix this error, apply the label `svirt_sandbox_file_t` to the directory `/var/lib/elastic/heartbeat/heartbeat-data/` on the Kubernetes node:
```

File: docs/advanced-topics/openshift.asciidoc [hunk 5]
```diff
@@ -226,3 +227,70 @@ elasticsearch-sample-es-6qk52mz5jk scc:restricted
...
+
+[source,shell]
+----
+chcon -Rt svirt_sandbox_file_t /var/lib/elastic/heartbeat/heartbeat-data/
+----
+
+This should be done on all the hosts where the heartbeat Pod can be deployed.
```
Comment (alaudazzi) [inline docs/advanced-topics/openshift.asciidoc:296]:
```suggestion
Repeat these steps on all the hosts where the heartbeat Pod can be deployed.
```

**Comment by barkbay** [comment]
@alaudazzi thanks for the review 🙇 
After some additional tests I added a small section [at the end of the document](https://github.com/elastic/cloud-on-k8s/pull/3588/files#diff-56b4ee636c261e49f541df3f94817569R298). I would be happy to have your review on this part too.

Thanks you

**Pull request merged**